src/printer.js

```diff
@@ -2291,6 +2291,7 @@ function printFunctionParams(path, print, options, expandArg) {
       parent.type === "TypeAlias") &&
     fun[paramsField].length === 1 &&
     fun[paramsField][0].name === null &&
+    fun[paramsField][0].typeAnnotation.type !== "UnionTypeAnnotation" &&
```

Comment:
I've been looking at the list and I think it makes more sense to have a whitelist rather than blacklist. Do you mind updating your code with the list below? Then we can merge it. Thanks!

Those are all the ones I could find that are just raw strings. Flow appears to not need parenthesis for a lot more but I think we should err on the safe side and add parenthesis for the rest.

```js
case "NullLiteralTypeAnnotation":
  return "null";
case "ThisTypeAnnotation":
  return "this";
case "NumberTypeAnnotation":
  return "number";
case "VoidTypeAnnotation":
  return "void";
case "NullTypeAnnotation":
  return "null";
case "EmptyTypeAnnotation":
  return "empty";
case "AnyTypeAnnotation":
  return "any";
case "MixedTypeAnnotation":
  return "mixed";
case "BooleanTypeAnnotation":
  return "boolean";
case "BooleanLiteralTypeAnnotation":
  return "" + n.value;
case "StringLiteralTypeAnnotation":
  return nodeStr(n, options);
```